scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala

```diff
@@ -775,18 +786,19 @@ object DiscoveryService {
           .take(config.kademliaAlpha)
           .toList
 
-        if (contacts.isEmpty)
-          Task.pure(closest)
-        else {
-          Task
-            .parTraverseUnordered(contacts)(fetchNeighbors)
-            .map(_.flatten.distinct)
-            .flatMap(bondNeighbors)
-            .flatMap { newNeighbors =>
-              val newClosest = (closest ++ newNeighbors).take(config.kademliaBucketSize)
-              val newAsked = asked ++ contacts
-              loop(local, newClosest, newAsked)
-            }
+        if (contacts.isEmpty) {
+          Task(logger.debug(s"Lookup finished for $target after asking ${asked.size} nodes.")).as(closest)
+        } else {
+          Task(logger.debug(s"Lookup for $target contacting ${contacts.size} nodes.")) >>
```

Comment:
Maybe we could also log how many nodes we already asked ?

Reply:
Sure thing.

Reply:
I thought we could log the addresses too but it's a bit verbose.